gadget/update.go

```diff
@@ -132,3 +200,97 @@ func canUpdateVolume(from *PositionedVolume, to *PositionedVolume) error {
 	}
 	return nil
 }
+
+type updatePair struct {
+	from *PositionedStructure
+	to   *PositionedStructure
+}
+
+func resolveUpdate(oldVol *PositionedVolume, newVol *PositionedVolume) (updates []updatePair) {
+	for j, oldStruct := range oldVol.PositionedStructure {
+		newStruct := newVol.PositionedStructure[j]
```

Comment:
The length of PositionedStructure for old and new are checked earlier by `canUpdateVolume` so this is fine, I wonder if we should have a comment about that here, or an early check on entering the function (error out with internal error if lengths are different).